pywatts_pipeline/core/pipeline.py

```diff
@@ -57,16 +60,110 @@ class Pipeline(BaseTransformer):
+            method=None):
+        """
+        Add a transformer or estimator to the pipeline.
+        :param estimator: A pyWATTS transformer or estimator
+        :type estimator: Base
+        :param name: The name of the transformer or estimator. It needs to be unique.
+        :type name: str
+        :param input_edges: A dict that maps the input argument for the fit/transform/predict method to the name
+                            of the predecessor. E.g. `{"argument_name": "PreviousStepName"}` maps the output of the step
+                            `"PreviousStepName"` to the argument "argument_name" if fit, transform or something similar is called.
+        :type input_edges: Dict[str, str]
+        :param condition: An optional callable argument that check if the step should be executed.
+        :param callbacks: Callbacks to use after results are processed.
+        :type callbacks: List[BaseCallback, Callable[[Dict[str, xr.DataArray]]]]
+        :param refit_conditions: A List of Callables of BaseConditions, which contains a condition that indicates if
+                                 the module should be trained or not
+        :type refit_conditions: List[Union[BaseCondition, Callable]]
+        :param computation_mode: Determines the computation mode of the step. Could be ComputationMode.Train,
+                                 ComputationMode.Transform, and Computation.FitTransform
+        :type computation_mode: ComputationMode
+        :param lag: Needed for online learning. Determines what data can be used for retraining.
+                    E.g., when 24 hour forecasts are performed, a lag of 24 hours is needed, else the retraining would
+                    use future values as target values.
+        :type lag: pd.Timedelta
+        :param method: Determines which method should be called when executing the transfomrer or estimator.
+                       E.g. `predict`, `transform`, or `inverse_transform`.
+        """
+        self.reset()
+        self._pipeline_construction_informations.append([
+            estimator, name, input_edges, {"callbacks": callbacks if not callbacks is None else [],
+                                           "condition": condition,
+                                           "computation_mode": computation_mode,
+                                           "refit_conditions": refit_conditions if not refit_conditions is None else [],
+                                           "lag": lag,
+                                           "method": method}
+        ])
+        return self._add(self._pipeline_construction_informations)[-1]
+
+    def _add(self, steps):
+        self.steps = {}
+        self.start_steps = {}
+        from pywatts_pipeline.core.steps.step_factory import StepFactory
+        step_informations = []
+        for transformer, name, input_edges, add_params in steps:
+            if id(transformer) not in self.est_dict:
```

Comment:
Do not use id(object) instead use the object directly as key. 

id(object) may cause problems if the object is destroyed. Then the id can be assigned to another object.

Reply:
See the following example
```
b = "test"
print(id(b))
del b
c = "test"
print(id(c))
```